sdks/python/container/Dockerfile

```diff
@@ -60,6 +60,7 @@ RUN mkdir -p /usr/local/gcloud && \
     curl -s -O https://dl.google.com/dl/cloudsdk/release/google-cloud-sdk.tar.gz && \
     tar -xf google-cloud-sdk.tar.gz && \
     /usr/local/gcloud/google-cloud-sdk/install.sh && \
+    rm -rf /usr/local/gcloud/google-cloud-sdk/.install/.backup && \
     rm google-cloud-sdk.tar.gz
```

Comment:
we do pip install again after `rm -rf /root/.cache/pip`. I think we can move that line to here

Reply:
@liferoad the point here is that if we move, `/root/.cache/pip` will be packaged on that layer, so it should increase the overall size.

Reply:
I mean this:
```
rm -rf /usr/local/gcloud/google-cloud-sdk/.install/.backup && \
rm rm -rf /root/.cache/pip &&\
```

Reply:
As @bvolpato reiterated, doing so would be suboptimal right now. Happy to discuss in person further.

Another reason,  for readability it may be better to remove relevant temp content roughly in the same place in the code where the temp content was added. 

That said, i sent out to reduce the number of layers and clear pip cache at the very end of the main layer https://github.com/apache/beam/pull/27006